### PR TITLE
make header values for WebSocket upgrade case insensitive

### PR DESCRIPTION
**Pull request by mbanikazemi**
This PR addresses the issue we ran into during the wWebSocket upgrade phase in our use case .

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -196,12 +196,17 @@ func isWebSocketUpgrade(header http.Header) (bool, error) {
 	if header.Get("Sec-Websocket-Protocol") != grpcwebsocket.SubprotocolName {
 		return false, nil
 	}
-	if header.Get("Connection") != "Upgrade" {
+
+	connectionValue := header.Get("Connection")
+	if (connectionValue != "upgrade") && (connectionValue != "Upgrade") {
```
Comment (connorgorman) [inline server/server.go:201]: I think 
```
!strings.EqualFold(connectionValue, "upgrade")
```
and similarly below would be ideal

**Review by connorgorman — COMMENTED** [review]
Thanks for the contribution! Added a small comment about using EqualFold instead

File: server/server.go (hunk 1 above)
Comment (RTann) [reply server/server.go:201]: I think I'd prefer `!strings.EqualFold(header.Get("Connection"), "upgrade")`, but either way is fine

Comment (mbanikazemi) [reply server/server.go]: Made changes as suggested.

**Review by RTann — APPROVED** [review]

**Review by connorgorman — APPROVED** [review]

**Comment by mbanikazemi** [comment]
Thanks. Unless there is a pre set release schedule, would you please consider cutting a new release when this is merged so we can get the changes from the new release.

**Comment by RTann** [comment]
> Thanks. Unless there is a pre set release schedule, would you please consider cutting a new release when this is merged so we can get the changes from the new release.

Done :)

**Pull request merged**